File: 0144-binary-tree-preorder-traversal/0144-binary-tree-preorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    ArrayList<Integer> values = new ArrayList<Integer>();
    public void traverse(TreeNode node){
        if (node != null){
            values.add(node.val);
            traverse(node.left);
            traverse(node.right);
        }
    }
    
    public List<Integer> preorderTraversal(TreeNode root) {
        traverse(root);
        return values;
    }
}

File: 0144-binary-tree-preorder-traversal/0144-binary-tree-preorder-traversal.py

class Solution:
    
    def traverse(self, node: TreeNode) -> None:
        if node:
            self.values.append(node.val)
            self.traverse(node.left)
            self.traverse(node.right)
        

    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        self.values = []
        self.traverse(root)
        return self.values

